app/modules/individuals/parameters.py

```diff
@@ -26,29 +29,67 @@ class Meta(schemas.DetailedIndividualSchema.Meta):
 
 class PatchIndividualDetailsParameters(PatchJSONParameters):
     # pylint: disable=abstract-method,missing-docstring
-    OPERATION_CHOICES = (PatchJSONParameters.OP_REPLACE,)
+    OPERATION_CHOICES = (
+        PatchJSONParameters.OP_REPLACE,
+        PatchJSONParameters.OP_ADD,
+        PatchJSONParameters.OP_REMOVE,
+    )
+
+    # Is this weird? I need to modify EDM and Houston records
+    PATH_CHOICES_HOUSTON = ['encounters']
+
+    PATH_CHOICES_EDM = ['encounters']
+
+    PATH_CHOICES = tuple(
+        '/%s' % field for field in (PATH_CHOICES_EDM + PATH_CHOICES_HOUSTON)
+    )
+
+    @classmethod
+    def remove(cls, obj, field, value, state):
+        has_permission = rules.ObjectActionRule(obj, AccessOperation.WRITE).check()
+        ret_val = False
+        if has_permission:
+            if field == 'encounters':
+                for encounter_guid in value:
+                    if is_valid_uuid(encounter_guid):
+                        from app.modules.encounters.models import Encounter
+
+                        encounter = Encounter.query.filter(
+                            Encounter.guid == encounter_guid
+                        ).first()
+                        if encounter is not None and encounter in obj.encounters:
+                            obj.remove_encounter(encounter)
+                            ret_val = True
+            else:
+                super(PatchIndividualDetailsParameters, cls).replace(
+                    obj, field, value, state
+                )
+        return ret_val
 
-    VALID_FIELDS = ['encounters']
-    PATH_CHOICES = tuple('/%s' % field for field in (VALID_FIELDS))
+    @classmethod
+    def add(cls, obj, field, value, state):
+        return cls.replace(obj, field, value, state)
 
     @classmethod
     def replace(cls, obj, field, value, state):
+
         ret_val = False
-        if field == 'encounters':
-            if len(value) == 0:
-                ret_val = True
-            encounter_list = []
-            for encounter_guid in value:
-                if is_valid_uuid(encounter_guid):
-                    from app.modules.encounters.models import Encounter
-
-                    encounter = Encounter.query.filter(
-                        Encounter.guid == encounter_guid
-                    ).first()
-                    if encounter is not None and encounter not in encounter_list:
-                        encounter_list.append(encounter)
-                    ret_val = True
-            obj.encounters = encounter_list
-        if ret_val is False:
-            super(PatchIndividualDetailsParameters, cls).replace(obj, field, value, state)
+        has_permission = rules.ObjectActionRule(obj, AccessOperation.WRITE).check()
+        if has_permission:
+            if field == 'encounters':
+                for encounter_guid in value:
+                    if is_valid_uuid(encounter_guid):
```

Comment:
likewise Encounter.query.get(encounter_guid) could replace quite a bit of this.

Reply:
As in comment above, changed for now.